presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -197,12 +196,37 @@ private ResourceGroupInfo getSummaryInfo()
                     runningTimeLimit,
                     queuedTimeLimit,
                     DataSize.succinctBytes(cachedMemoryUsageBytes),
-                    queuedQueries.size() + descendantQueuedQueries,
-                    runningQueries.size() + descendantRunningQueries,
+                    getQueuedQueries(),
+                    getRunningQueries(),
                     eligibleSubGroups.size(),
                     subGroups.values().stream()
-                            .map(InternalResourceGroup::getSummaryInfo)
-                            .collect(toImmutableList()));
+                            .filter(group -> group.getRunningQueries() + group.getQueuedQueries() > 0)
+                            .map(InternalResourceGroup::getChildInfo)
+                            .collect(toImmutableList()),
+                    null);
+        }
+    }
+
+    private ResourceGroupInfo getChildInfo()
```

Comment:
The function name is misleading. It's still resource group info of the current `this` resource group, not its child.

Reply:
Good point.